src/cli/build_command.zig

```diff
@@ -496,7 +519,15 @@ pub const BuildCommand = struct {
             }
 
             for (output_files) |f| {
-                f.writeToDisk(root_dir, from_path) catch |err| {
+                // When using --no-bundle with --outdir, we want to place files directly in the output directory
+                // without recreating the source directory structure
+                var write_from_path = from_path;
+                if (this_transpiler.options.transform_only and outdir.len > 0) {
+                    // Use the output directory as the from_path to avoid recreating the source directory structure
```

Comment:
why is this desired? i think we do want to construct the same directory structure?

```
touch a.ts 
mkdir dir
touch dir/b.ts
bun build a.ts dir/b.ts --outdir=out

Bundled 2 modules in 2ms

  a.js     63 bytes  (entry point)
  dir/b.js 67 bytes  (entry point)
```